js/parts/Chart.js

```diff
@@ -433,6 +433,29 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
         }
         // Adjust title layout (reflow multiline text)
         chart.layOutTitles();
+        // link stacked series
+        i = series.length;
+        while (i--) {
+            serie = series[i];
+            if (serie.options.stacking) {
+                hasStackedSeries = true;
+                if (serie.isDirty) {
+                    hasDirtyStacks = true;
+                    break;
+                }
+            }
+        }
+        if (hasDirtyStacks) { // mark others as dirty
+            i = series.length;
+            while (i--) {
+                serie = series[i];
+                if (serie.options.stacking &&
+                    // These series have their own dirting logic
+                    !serie.type.match(/column|columnrange|bar/g)) {
+                    serie.isDirty = true;
```

Comment:
Could we set the `serie.isDirty = true` flag in 444 line and avoid double while loop?

Reply:
In short: this is the old logic for series other than column, columnrange, bar. We need to loop one more time separately to dirty all series, not only the ones found after first dirty.
@TorsteinHonsi will know more.